addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +113,144 @@ def _get_duration_from_tracking(self, trackings):
+            def _get_rotting_domain(self):
+                return super()._get_rotting_domain() + [
+                    ('won_status', '=', 'pending'),
+                    ('type', '=', 'opportunity')
+                ]
+
+        Note that the fields present in _get_rotting_domain() are the same as the ones present in _get_rotting_depends_fields()
+
+    - The is_rotting, rotting_days fields need to be added to the relevant views
+        (as well as the "rotting_threshold_days" field on the tracked model).
+        You may want to use the 'rotting' field widget to display
+        the fields visually on form and kanban view
+        (please note- these widgets need both rotting_days and is_rotting fields on the view to function).
+
+    If the rotting_threshold_days field is not defined on the tracked module,
+    or if the value of rotting_threshold_days is 0,
+    then the resource will never rot.
+    """
+
+    rotting_days = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting')
+
+    # To override in inheriting models
+    def _get_rotting_depends_fields(self):
+        if self._is_rotting_enabled():
+            return ['date_last_stage_update', f'{self._track_duration_field}.rotting_threshold_days']
+        return []
+
+    # To override in inheriting models
+    def _get_rotting_domain(self):
+        if self._is_rotting_enabled():
+            return [
+                (f'{self._track_duration_field}.rotting_threshold_days', '!=', 0),
+            ]
+        return [fields.Domain.FALSE]
+
+    def _is_rotting_enabled(self):
+        return hasattr(self, '_track_duration_field') and 'rotting_threshold_days' in self[self._track_duration_field] and (
+            not self  # api.model call
+            or any(stage.rotting_threshold_days for stage in self[self._track_duration_field])
+        ) and 'date_last_stage_update' in self
+
+    @api.depends(lambda self: self._get_rotting_depends_fields())
+    def _compute_rotting(self):
+        could_rot_records = self.filtered_domain(self._get_rotting_domain())
+        clean_records = (self - could_rot_records)
+        for stage, records in could_rot_records.grouped(self._track_duration_field).items():
+            rotting_records = records.filtered(lambda record: fields.Datetime.now() > record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days))
+            rotting_records.is_rotting = True
+            for record in rotting_records:
+                record.rotting_days = (fields.Datetime.now() - record.date_last_stage_update).days
+            clean_records += (records - rotting_records)
+        clean_records.is_rotting = False
+        clean_records.rotting_days = 0
```

Comment:
Suggesting some linting / naming, also use self.env.cr.now() instead of datetime.now, remove nondeterministic behavior
```suggestion
    def _compute_rotting(self):
        now = self.env.cr.now()
        rot_enabled = self.filtered_domain(self._get_rotting_domain())
        other = self - could_rot_records
        for stage, records in rot_enabled.grouped(self._track_duration_field).items():
            rotting = records.filtered(
                lambda record: record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days) < now
            )
            for record in rotting_records:
                record.is_rotting = True
                record.rotting_days = (now - record.date_last_stage_update).days
            other += records - rotting_records
        others.update({
                'is_rotting': False,
                'rotting_days': 0,
            })
```